lib/SQL/Translator/Diff.pm

```diff
@@ -99,6 +99,88 @@ sub BUILD {
   }
 }
 
+# This function detects changes between two versions. Four scenarios are possible:
+# v1  | v2
+# X   | Y:RX | It was renamed to Y from X   - rename
+# X   | X*   | Something changed inside     - alter
+# -   | X    | No field in the old version  - create
+# X   | -    | No field in the new version  - drop
+
+# For each of this scenario corresponding callback is fired: rename, alter, drop, create.
+# Additionally 'next' callback if fired for every comparison. It could be used to prepare
+# data structures where to fill the comparison/diff result.
+# The callbacks are called with the next parameters:
+# rename( $src_version, $dst_version )
+# alter ( $src_version, $dst_version )
+# create( $dst_version )
+# drop  ( $src_version )
+# next  ( $dst_name, $dst_version )
```

Comment:
i'm not understanding what the `next` callback is meant to do; would you mind expounding on that?

Reply:
`next` call back is fired for every `dst` column. This is to run some code for each `dst` name/column/table.

At the moment this call back is used to prepare/initialize required data structures which will be filled/used later.
![image](https://github.com/user-attachments/assets/13a50523-a8a2-4214-a601-d19f5d7b4799)

I agree that the name is not the best. Probably it could be better if we name it `every` or `init`. Just suggest better name and I'll change it.